nomad/grapl-core.nomad

```diff
@@ -0,0 +1,428 @@
+# For rust logging output
+variable "RUST_LOG" {
+    type = string
+    default = "INFO"
+}
+
+# Which tag to use for grapl-specific images
+variable "tag" {
+    type = string
+    default = "latest"
+}
+
+# The container registry that we can find grapl-specific services
+variable "container_registry" {
+    type = string
+    default = "localhost:5000"
+}
+
+variable "aws_region" {
+    type = string
+    default = "us-west-2"
+}
+
+variable "aws_sqs_url" {
+    type = string
+}
+
+variable "aws_account_id" {
+    type = string
+    default = "000000000000"
+}
+
+variable "dgraph_replicas" {
+    type = number
+    default = 1
+}
+
+variable "dgraph_shards" {
+    type = number
+    default = 1
+}
+
+# Where is redis located?
+variable "redis_endpoint" {
+    type = string
+}
+
+# What is the name of the schema table?
+variable "schema_table_name" {
+    type = string
+}
+
+# https://github.com/grapl-security/grapl/blob/af6f2c197d52e9941047aab813c30d2cbfd54523/pulumi/infra/dynamodb.py#L118
+# Without deployment name
+variable "session_table" {
+    type = string
+    default = "dynamic_session_table"
+}
+
+# How many graph mergers should be running in tandem
+variable "graph_mergers" {
+    type = number
+    default = 1
+}
+
+# How many node identifiers should be running in tandem
+variable "node_identifiers" {
+    type = number
+    default = 1
+}
+
+variable "node_identifiers_retry" {
+    type = number
+    default = 1
+}
+
+locals {
+    # DGraph Alphas (shards * replicas)
+    dgraph_alphas = [for alpha_id in range(0, var.dgraph_replicas * var.dgraph_shards): {
+        id: alpha_id,
+        grpc_private_port: alpha_id + 7080,
```

Comment:
This is admittedly _super_ nitpicky, but having the "base number" come first, and the variable second reads more naturally (to me, at any rate 😅)

e.g.,
```
7080 + 1
7080 + 2
7080 + 3
```
vs
```
1 + 7080
2 + 7080
3 + 7080
```

Reply:
Also, why is it necessary for each shard to use a different port? They should be independent, no?

If this is to support local usecases, aren't there other ways of mapping concrete ports from the host to the individual service?

Reply:
Also, should these base numbers be extracted to variables / locals? I see they're used in multiple places, and magic numbers make me nervous.

Reply:
> This is admittedly super nitpicky, but...

Good point :) I'll change that!

> why is it necessary for each shard to use a different port? They should be independent, no?

Ideally, yes! However, we need to use different ports due to the way that consul connect works. In consul connect land, different port == different upstream. It's *probably* possible to make it so that the local port is the same in every instance, but seeing as the `service`s would have to operate on different ports it seemed fine fine to use different ports.

> Also, should these base numbers be extracted to variables / locals?

Good idea on the locals! I can move them over :)

Reply:
> Ideally, yes! However, we need to use different ports due to the way that consul connect works. In consul connect land, different port == different upstream. It's probably possible to make it so that the local port is the same in every instance, but seeing as the services would have to operate on different ports it seemed fine fine to use different ports.

I need to get more familiar with Nomad to fully grok this, but for now it's probably fine as it is. I'm sure all this stuff will be evolving anyway 😄